src/pybind/mgr/dashboard/frontend/src/app/ceph/performance-counter/performance-counter/performance-counter.component.spec.ts

```diff
@@ -1,38 +1,19 @@
-import { Component, Input } from '@angular/core';
+import { HttpClientTestingModule } from '@angular/common/http/testing';
 import { ComponentFixture, TestBed } from '@angular/core/testing';
 import { RouterTestingModule } from '@angular/router/testing';
 
-import { PerformanceCounterService } from '../../../shared/api/performance-counter.service';
+import { SharedModule } from '../../../shared/shared.module';
 import { configureTestBed } from '../../../shared/unit-test-helper';
+import { TablePerformanceCounterComponent } from '../table-performance-counter/table-performance-counter.component';
```

Comment:
Do we really want to include nested components?
Angular suggest we use a stub component and/or `NO_ERRORS_SCHEMA` [1]

Same applies to other files in this PR.

[1] https://angular.io/guide/testing#nested-component-tests

Reply:
If we create an extra class that has to be updated as soon as the input changes of `TablePerformanceCounter` - Someone will forget about it and than we have a false positive test result and a bug in the dashboard.
Mock components make sense if the thing you test can be used in any component, a good example for this would be the deletion dialog test suite.

Also we haven't even written any test for it, so we should not mock things out without knowing what we will be testing in the future.

If a mock is required for a specific test case that's fine, but not before the test. Also we should only try to mock external things like included libraries or API calls.